src/components/PeopleTable.tsx

```diff
@@ -1,644 +1,139 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+
+import React from 'react';
+import { Person } from '../types';
+import classNames from 'classnames';
+import { useParams, Link, useSearchParams } from 'react-router-dom';
+import { PersonLink } from './PersonLink';
+import { ColumnOption } from '../types/ColumnOption';
+
+type Props = {
+  people: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { slug } = useParams<{ slug: string }>();
+
+  const [searchParams] = useSearchParams();
+  const sort = searchParams.get('sort') || null;
+  const order = searchParams.get('order') || 'asc';
+  const sexFilter = searchParams.get('sex');
+
+  const findPersonByName = (name?: string) => {
+    return name ? people.find(person => person.name === name) : undefined;
+  };
+
+  const filteredPeople = sexFilter
+    ? people.filter(person => person.sex.toLowerCase() === sexFilter)
+    : people;
+
+  const optionsForColumns = Object.values(ColumnOption);
+
   return (
     <table
       data-cy="peopleTable"
       className="table is-striped is-hoverable is-narrow is-fullwidth"
     >
       <thead>
         <tr>
-          <th>
-            <span className="is-flex is-flex-wrap-nowrap">
-              Name
-              <a href="#/people?sort=name">
-                <span className="icon">
-                  <i className="fas fa-sort" />
+          {optionsForColumns.map(option => {
+            const isSortedColumn = sort === option.toLowerCase();
+            const newOrder = isSortedColumn && order === 'asc' ? 'desc' : 'asc';
+            const isSortable =
+              option !== ColumnOption.MOTHER && option !== ColumnOption.FATHER;
+
+            return (
+              <th key={option}>
+                <span className="is-flex is-flex-wrap-nowrap">
+                  {option}
+                  {isSortable && (
+                    <Link
+                      to={`/people?sort=${option.toLowerCase()}&order=${newOrder}`}
```

Comment:
The sorting link uses a hardcoded path `/people`. Ensure that this path is correct and matches the route configuration in your application. If the base path changes, this will need to be updated.